nemo/core/classes/modelPT.py

```diff
@@ -471,6 +473,9 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
                     optim_config['sched']['t_num_workers'] = self._trainer.num_processes * self._trainer.num_nodes
                 elif self._trainer.accelerator == "ddp":
                     optim_config['sched']['t_num_workers'] = self._trainer.num_gpus * self._trainer.num_nodes
+                elif isinstance(self._trainer.accelerator.training_type_plugin, NLPDDPPlugin):
+                    app = AppState()
+                    optim_config['sched']['t_num_workers'] = app.data_parallel_size
```

Comment:
The t_num_workers should be the data parallel size in the presence of model parallelism workers. Otherwise, the `max_step` calculation for optim scheduler will be off

Reply:
This is fine from my side.